src/main/java/me/petterim1/welcomeform/Main.java

```diff
@@ -77,39 +77,44 @@ public void run() {
                         showForm(p);
                     }
                 }
-            }.runTaskLater(this, delay);
+            }.runTaskLater(this, secondsAfterJoin);
         }
     }
 
     @EventHandler(ignoreCancelled = true)
     public void onFormResponse(PlayerFormRespondedEvent e) {
-        if (!button || !buttonAction) return;
+        if (!showButton || !buttonAction) return;
         if (e.getResponse() == null) return;
         if (e.getWindow().wasClosed()) return;
         if (e.getWindow() instanceof FormWindowSimple) {
-            if (formTitle.equals(((FormWindowSimple) e.getWindow()).getTitle())) {
-                if (buttonText.equals(((FormWindowSimple) e.getWindow()).getResponse().getClickedButton().getText())) {
-                    getServer().dispatchCommand(e.getPlayer(), buttonCommand);
+            Player p = e.getPlayer();
+            if (placeholderFunc.apply(formTitle, p).equals(((FormWindowSimple) e.getWindow()).getTitle())) {
+                if (placeholderFunc.apply(buttonText, p).equals(((FormWindowSimple) e.getWindow()).getResponse().getClickedButton().getText())) {
+                    getServer().dispatchCommand(p, buttonCommand.replace("%player%", "\"" + p.getName()) + "\"");
                 }
             }
         }
     }
 
     private void showForm(Player p) {
-        if (api != null) {
-            formTitle = api.translateString(formTitle, p);
-            formText = api.translateString(formText, p);
-            buttonText = api.translateString(buttonText, p);
-            FormWindowSimple form = new FormWindowSimple(formTitle, formText);
-            if (button) form.addButton(new ElementButton(buttonText));
-            p.showFormWindow(form);
-        }
+        FormWindowSimple form = new FormWindowSimple(placeholderFunc.apply(formTitle, p), placeholderFunc.apply(formText, p));
+        if (showButton) form.addButton(new ElementButton(placeholderFunc.apply(buttonText, p)));
+        p.showFormWindow(form);
     }
 
     private void initPlaceholders() {
         try {
-            api = PlaceholderAPI.getInstance();
-        } catch (Throwable ignored) {
+            Class<?> placeholderAPI = Class.forName("com.creeperface.nukkit.placeholderapi.api.PlaceholderAPI");
```

Comment:
If not using reflection could we do this only once on plugin startup? Or rather check if the PlaceholderAPI plugin is loaded

Reply:
Maybe so?